src/ctia/schemas/graphql/helpers.clj

```diff
@@ -372,17 +395,20 @@
    ^String description
    type-resolver-fn
    types]
-  (let [type-resolver (fn->type-resolver type-resolver-fn)
-        graphql-union (-> (GraphQLUnionType/newUnionType)
-                          (.description description)
-                          (.name union-name)
-                          ; FIXME: this method is deprecated
-                          (.typeResolver type-resolver))]
-    (doseq [type types]
-      (if (instance? GraphQLObjectType type)
-        (.possibleType graphql-union ^GraphQLObjectType type)
-        (.possibleType graphql-union ^GraphQLTypeReference type)))
-    (.build graphql-union)))
+    (get-or-update-type-registry
+      default-type-registry
+      union-name
+      #(let [type-resolver (fn->type-resolver type-resolver-fn)
+             graphql-union (-> (GraphQLUnionType/newUnionType)
+                               (.description description)
+                               (.name union-name)
+                               ; FIXME: this method is deprecated
+                               (.typeResolver type-resolver))]
+         (doseq [type types]
+           (if (instance? GraphQLObjectType type)
+             (.possibleType graphql-union ^GraphQLObjectType type)
+             (.possibleType graphql-union ^GraphQLTypeReference type)))
+         (.build graphql-union))))
```

Comment:
Please temporarily permit the funny indentation, a forthcoming commit will wrap this entire return value in a function and restore some order.